api/client/registry/login.go

```diff
@@ -52,7 +53,16 @@ func runLogin(dockerCli *client.DockerCli, opts loginOptions) error {
 
 	var serverAddress string
 	var isDefaultRegistry bool
-	if opts.serverAddress != "" {
+
+	// when a user supplied registry url is the default docker.io registry just ignore it
+	// without this check the login will succed, but the credentials will not be saved properly and can't be used
+	// handle urls without schema for backward compatibility and pass the integration tests
+	u, _ := url.Parse(opts.serverAddress)
+	if u.Host == "" {
```

Comment:
I'm uncomfortable with skipping the error check. I think checking for an error is a better indication than whether `u.Host` is empty. I believe it would be completely legitimate for `url.Parse` to return garbage in the `Host` field as long as it returned an error alongside it.